controllers/kuadrant_controller.go

```diff
@@ -495,6 +495,15 @@ func (r *KuadrantReconciler) reconcileAuthorino(ctx context.Context, kObj *kuadr
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *KuadrantReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	ok, err := kuadrantgatewayapi.IsGatewayAPIInstalled(mgr.GetRESTMapper())
+	if err != nil {
+		return err
+	}
+	if !ok {
+		r.Logger().Info("Kuadrant controller disabled. GatewayAPI was not found")
+		return nil
+	}
+
```

Comment:
This is a great point. I would try to avoid the restart if we can. If we are to keep the pod running anyway, at least it could retry the conditions in a loop, setup the reconcilers automatically in case the state changes.

Reply:
Yeah, ideally the controller should keep trying until it finds it.

I thought about implementing it. But I wanted to discuss first the implementation

Reply:
It can be done in another PR.